src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final byte DATE_POS = 0;
+    private static final byte NAME_POS = 1;
+    private static final byte HOURS_POS = 2;
+    private static final byte SALARY_POS = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int periodSalary;
+        StringBuilder result = new StringBuilder();
+
+        result.append("Report for period ")
+                .append(LocalDate.parse(dateFrom, DATE_FORMAT).format(DATE_FORMAT))
+                .append(" - ").append(LocalDate.parse(dateTo, DATE_FORMAT).format(DATE_FORMAT));
+
+        for (String name: names) {
+            periodSalary = 0;
```

Comment:
```suggestion
        int periodSalary = 0;
```
його можна одразу тут створювати так як ми тільки в циклі цю змінну використовуємо